astropy/coordinates/representation/base.py

```diff
@@ -143,10 +145,26 @@ class BaseRepresentationOrDifferential(MaskableShapedLikeNDArray):
 
     # Ensure multiplication/division with ndarray or Quantity doesn't lead to
     # object arrays.
-    __array_priority__ = 50000
+    __array_priority__: ClassVar[int] = 50000
+
+    name: ClassVar[str]
+    """Name of the representation or differential.
+
+    In lower case, with any trailing 'representation' or 'differential' removed. (E.g.,
```

Comment:
I'd make this `By default, the name of the class with ...`

Reply:
SGTM!

Reply:
Done.